packages/aws-rfdk/lib/core/lambdas/nodejs/version-provider/test/version-provider.test.ts

```diff
@@ -0,0 +1,187 @@
+  expect(linuxInstallerVersion).not.toBeNull();
+
+  if (result === null) { return; }
+  expect(linuxInstallerVersion?.Uris).toEqual({
+    bundle: 's3://thinkbox-installers/Deadline/10.1.9.2/Linux/Deadline-10.1.9.2-linux-installers.tar',
+    clientInstaller: 's3://thinkbox-installers/Deadline/10.1.9.2/Linux/DeadlineClient-10.1.9.2-linux-x64-installer.run',
+    repositoryInstaller: 's3://thinkbox-installers/Deadline/10.1.9.2/Linux/DeadlineRepository-10.1.9.2-linux-x64-installer.run',
+  });
+  expect(linuxInstallerVersion?.MajorVersion).toEqual('10');
+  expect(linuxInstallerVersion?.MinorVersion).toEqual('1');
+  expect(linuxInstallerVersion?.ReleaseVersion).toEqual('9');
+  expect(linuxInstallerVersion?.PatchVersion).toEqual('2');
+
+  const windowsInstallerVersion = result.get(Platform.windows);
+  expect(windowsInstallerVersion).not.toBeNull();
+
+  if (result === null) { return; }
+  expect(windowsInstallerVersion?.Uris).toEqual({
+    bundle: 's3://thinkbox-installers/Deadline/10.1.8.5/Windows/Deadline-10.1.8.5-windows-installers.zip',
+    clientInstaller: 's3://thinkbox-installers/Deadline/10.1.8.5/Windows/DeadlineClient-10.1.8.5-windows-installer.exe',
+    repositoryInstaller: 's3://thinkbox-installers/Deadline/10.1.8.5/Windows/DeadlineRepository-10.1.8.5-windows-installer.exe',
+  });
+  expect(windowsInstallerVersion?.MajorVersion).toEqual('10');
+  expect(windowsInstallerVersion?.MinorVersion).toEqual('1');
+  expect(windowsInstallerVersion?.ReleaseVersion).toEqual('8');
+  expect(windowsInstallerVersion?.PatchVersion).toEqual('5');
+});
+
+test('validate input', async () => {
+  expect(versionProvider.implementsIVersionProviderProperties({
+    product: Product.deadline,
+    versionString: '10.1.9.2',
+    platform: 'linux',
+  })).toBeTruthy();
+
+  expect(versionProvider['implementsIVersionProviderProperties']('test')).toEqual(false);
+
+  expect(versionProvider.implementsIVersionProviderProperties({
+    versionString: 'version',
+  })).toEqual(false);
+
+  expect(versionProvider.implementsIVersionProviderProperties({
+    product: Product.deadline,
+    versionString: 'version',
+  })).toEqual(false);
+
+  expect(versionProvider['implementsIVersionProviderProperties']({
+    product: Product.deadline,
+    platform: 'test',
+  })).toEqual(false);
+});
+
+test('invalide file path', () => {
+  expect(() => (new VersionProvider())['readInstallersIndex']()).toThrowError(/File path should be defined./);
+  expect(() => (new VersionProvider('test.txt'))['readInstallersIndex']()).toThrowError(/File test.txt was not found/);
+});
+
+test('get latest version', () => {
+  expect(() => versionProvider['getLatestVersion']('linux',{})).toThrowError(/Information about latest version can not be found/);
+  expect(() => versionProvider['getLatestVersion']('linux',{latest: {}})).toThrowError(/Information about latest version for platform linux can not be found/);
+});
+
+test('get requested Uri version', () => {
```

Comment:
I would change this to use `test.each()` with the version string and expected output as inputs.